torchgeo/datasets/solar_plants_brazil.py

```diff
@@ -0,0 +1,230 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""SolarPlantsBrazil dataset."""
+
+import glob
+import os
+from collections.abc import Callable
+from typing import Literal
+
+import matplotlib.pyplot as plt
+import numpy as np
+import rasterio
+import torch
+from matplotlib.figure import Figure
+from torch import Tensor
+
+from .errors import DatasetNotFoundError
+from .geo import NonGeoDataset
+from .utils import Path, download_and_extract_archive
+
+
+class SolarPlantsBrazil(NonGeoDataset):
+    """Solar Plants Brazil dataset (semantic segmentation for photovoltaic detection).
+
+    The `Solar Plants Brazil <https://huggingface.co/datasets/FederCO23/solar-plants-brazil>`__
+    dataset provides satellite imagery and pixel-level annotations for detecting photovoltaic
+    solar power stations.
+
+    Dataset features:
+
+    * 272 RGB+NIR GeoTIFF images (256x256 pixels)
+    * Binary masks indicating presence of solar panels (1 = panel, 0 = background)
+    * Organized into `train`, `val`, and `test` splits
+    * Float32 GeoTIFF files for both input and mask images
+    * Spatial metadata included (CRS, bounding box), but not used directly for training
+
+    Folder structure:
+
+    .. code-block:: text
+
+        root/train/input/img(123).tif
+        root/train/labels/target(123).tif
+
+    Access:
+
+    * Dataset is hosted on Hugging Face: https://huggingface.co/datasets/FederCO23/solar-plants-brazil
+    * Code and preprocessing steps available at: https://github.com/FederCO23/UCSD_MLBootcamp_Capstone
+
+    .. versionadded:: 0.8
+
+    """
+
+    url = 'https://huggingface.co/datasets/FederCO23/solar-plants-brazil/resolve/main/solarplantsbrazil.zip'
```

Comment:
```suggestion
    url = 'https://huggingface.co/datasets/FederCO23/solar-plants-brazil/resolve/1dc13a453ef6acabf08a1781c523fd1db3d9bcc5/solarplantsbrazil.zip'
```
Use a stable commit hash so that the MD5 doesn't change even if the dataset is later updated.